src/Voron/Data/Tables/TableValueReader.cs

```diff
@@ -1,48 +1,112 @@
-﻿// -----------------------------------------------------------------------
+// -----------------------------------------------------------------------
 //  <copyright file="a.cs" company="Hibernating Rhinos LTD">
 //      Copyright (c) Hibernating Rhinos LTD. All rights reserved.
 //  </copyright>
 // -----------------------------------------------------------------------
 
 using System;
 using System.Diagnostics;
-using System.Diagnostics.CodeAnalysis;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
 using System.Text;
+using Sparrow;
 using Sparrow.Compression;
 
 namespace Voron.Data.Tables
 {
-    public readonly unsafe struct TableValueReader
+    public unsafe interface ITableValueReader
```

Comment:
Why do we need an interface here?

Reply:
For the `N1`, `N2`, etc?

Reply:
Yes, that's how we achieve code specialization.